nvm.sh

```diff
@@ -759,15 +759,37 @@ nvm_ls_remote_index_tab() {
 
 nvm_checksum() {
   local NVM_CHECKSUM
-  if nvm_has "sha1sum" && ! nvm_is_alias "sha1sum"; then
-    NVM_CHECKSUM="$(command sha1sum "$1" | command awk '{print $1}')"
-  elif nvm_has "sha1" && ! nvm_is_alias "sha1"; then
-    NVM_CHECKSUM="$(command sha1 -q "$1")"
-  elif nvm_has "shasum" && ! nvm_is_alias "shasum"; then
-    NVM_CHECKSUM="$(shasum "$1" | command awk '{print $1}')"
+  if [ -z "$3" ] || [ "$3" == "sha1" ]; then
+    if nvm_has "sha1sum" && ! nvm_is_alias "sha1sum"; then
+      NVM_CHECKSUM="$(command sha1sum "$1" | command awk '{print $1}')"
+    elif nvm_has "sha1" && ! nvm_is_alias "sha1"; then
+      NVM_CHECKSUM="$(command sha1 -q "$1")"
+    elif nvm_has "shasum" && ! nvm_is_alias "shasum"; then
+      NVM_CHECKSUM="$(shasum "$1" | command awk '{print $1}')"
+    else
+      echo "Unaliased sha1sum, sha1, or shasum not found." >&2
+      return 2
+    fi
   else
-    echo "Unaliased sha1sum, sha1, or shasum not found." >&2
-    return 2
+    if nvm_has "sha256sum" && ! nvm_is_alias "sha256sum"; then
+      NVM_CHECKSUM="$(sha256sum "$1" | awk '{print $1}')"
+    elif nvm_has "shasum" && ! nvm_is_alias "shasum"; then
+      NVM_CHECKSUM="$(shasum -a 256 "$1" | awk '{print $1}')"
+    elif nvm_has "sha256" && ! nvm_is_alias "sha256"; then
+      NVM_CHECKSUM="$(sha256 -q "$1" | awk '{print $1}')"
+    elif nvm_has "gsha256sum" && ! nvm_is_alias "gsha256sum"; then
+      NVM_CHECKSUM="$(gsha256sum "$1" | awk '{print $1}')"
+    elif nvm_has "openssl" && ! nvm_is_alias "openssl"; then
+      NVM_CHECKSUM="$(openssl dgst -sha256 "$1" | rev | awk '{print $1}' | rev)"
+    elif nvm_has "libressl" && ! nvm_is_alias "libressl"; then
+      NVM_CHECKSUM="$(libressl dgst -sha256 "$1" | rev | awk '{print $1}' | rev)"
+    elif nvm_has "bssl" && ! nvm_is_alias "bssl"; then
+      NVM_CHECKSUM="$(bssl sha256sum "$1" | awk '{print $1}')"
+    else
+      echo "Unaliased sha256sum, sha256, or gsha256sum not found." >&2
```

Comment:
Thanks, I've updated this list in the most recent commit (https://github.com/creationix/nvm/commit/f73bfb65780709b181cc4f53bff554f67f26a2b3).